### ISSUE TITLE
Changed X11 RootWindow size and top location in macOS Sonoma results in menu bar overlapping

### ISSUE THREAD

**Issue by jens-maus** [XQuartz/XQuartz#364]
Since update to macOS Sonoma my dwm-based desktop running via Xquartz seem to have different X11 RootWindow settings resulting in the dwm windows to be expanded too far in the top direction so that the macOS menubar overlaps the X11 windows to some extend.

See here:
<img width="492" alt="Bildschirmfoto 2023-09-27 um 22 31 28" src="https://github.com/XQuartz/XQuartz/assets/5887769/793adb4c-b6f1-4563-989a-4d4dc66269f9">

Please note that along the bottom of the menubar there should be a red horizontal line marking the top of the X11 RootWindow which dwm usually occupies altogether. However, the text within the top urxvt terminal window is overlapped so that the text at the very top cannot be read due to the menubar overlapping.

The same happens with the default quartz-wm windows manage when one tries to move the an X11 windows to the very top:

<img width="538" alt="Bildschirmfoto 2023-09-27 um 22 32 29" src="https://github.com/XQuartz/XQuartz/assets/5887769/89125b5d-7ff5-46cb-bbb2-4fed6b468aa5">

Note the red, yellow and green button being overlapped by the macOS menubar as well as the text of the window bar not being readable.

When running `xdpyinfo` and `xrandr` the following output is shown:

```
$ xdpyinfo
...
screen #0:
  dimensions:    1728x1093 pixels (457x289 millimeters)
  resolution:    96x96 dots per inch
  depths (7):    24, 1, 4, 8, 15, 16, 32

$ xrandr
xrandr: Failed to get size of gamma for output default
Screen 0: minimum 1728 x 1093, current 1728 x 1093, maximum 3456 x 2234
default connected 1728x1093+0+0 0mm x 0mm
```

As one can see the height is output as 1093 pixels which as smaller in macOS ventura.

Thus, for some reason with macOS Sonoma Xquartz is using and reporting a larger X11 RootWindow size and top+left minimum location resulting in X11 windows to be able to be dragged over the border of the top menubar, thus making them partly overlapping so that some parts of the X11 window or text will not be visible again.

So is there any explanation for that or a workound using `xrandr` or similar that would instruct Xquartz to start with a different top location for the root window?

Please note, that the affected macOS system is a MacBookPro M1 system with the famous "Top Nodge". Other systems not having that top nodge does not seem to be affected by this issue. Thus, this could be actually related to the somewhat larger menu bar size compared to non-top-nodge mac systems. Thus, the additional offset due to the slightly larger menu bar might not be incorporated in the RootWindow size/position calculations, thought they might have been changed from Ventura to Sonoma...

**Comment by trolin522581** [comment]
Seeing the same on Sonoma on a MBP M2.

**Comment by jeremyhu** [comment]
We probably have "22" hardcoded somewhere for the menubar when calculating the drawable area..

**Comment by jens-maus** [comment]
After having updated to the latest macOS Sonoma 14.2 version the issue seems to be somehow resolved. At least I cannot drag any X window inside the top nodge menu bar anymore to see the same overlapping as mentioned in the initial report text above.

In fact, looking at the output of `xdpyinfo` and `xrandr` the X11 Root Window height values seem to be slightly reduced now by 13 pixels compared to my initial bug report in late september:

```
$ xdpyinfo
...
screen #0:
  dimensions:    1728x1080 pixels (457x286 millimeters)
  resolution:    96x96 dots per inch
  depths (7):    24, 1, 4, 8, 15, 16, 32
...

$ xrandr
xrandr: Failed to get size of gamma for output default
Screen 0: minimum 1728 x 1080, current 1728 x 1080, maximum 3456 x 2234
default connected 1728x1080+0+0 0mm x 0mm
```

As can be seen, the RootWindow height is now reported as 1080 while with earlier macOS Sonoma versions it was reported as 1093 for the same system, which perfectly explains why windows were suddenly overlapping with the top menu bar in early macOS Sonoma versions.

Thus, it seems that this was actually an issue/bug in macOS Sonoma itself and Apple has fixed it with the release of 14.2.

**Comment by jeremyhu** [comment]
Thanks
